### ISSUE TITLE
Exporting Union Types in Elm Syntax

### ISSUE THREAD

**Issue by trotha01** [elm-lang/elm-lang.org#523]
If we have a submodule with:
`type Animal = Cat | Dog`

we can export Cat and Dog with:
`module Submodule (Animal(..)) where`

and import with
`import Submodule exposing (Animal(..))`

Can we add something that explains this on the [syntax](http://elm-lang.org/docs/syntax) page

**Comment by jvoigtlaender** [comment]
Isn't there an issue or even a pull request open about this already?

**Comment by jvoigtlaender** [comment]
I guess I was thinking of this: https://github.com/elm-lang/elm-lang.org/pull/512. It's about something else to do with module documentation, though.

**Comment by halfzebra** [comment]
It is mentioned here [Elm Syntax — Modules](http://elm-lang.org/docs/syntax#modules) already, but I agree, it is not explicit. The best I could think of is adding a commentary about the presence of open imports for Union Types in Modules section.

**Comment by francescortiz** [comment]
I think it should be mentioned here too: https://guide.elm-lang.org/types/union_types.html

**Comment by granmoe** [comment]
Yeah, this is mysterious to a beginner and not emphasized enough in the docs (or at least not enough that I remembered it). This github issue saved me :-)

**Comment by daniel-kun** [comment]
Glad I found this github issue, because I couldn't deduct this from [Elm Syntax - Modules](http://elm-lang.org/docs/syntax#modules)

**Comment by zmijunkie** [comment]
Puh I am so happy to have seen this issue... Now all my Types go into a single Types.elm

**Comment by ian-moore** [comment]
This was a good 30 minutes of headache for me... glad this issue is the first result when you google it.

**Comment by fiatjaf** [comment]
For anyone looking here. Export your union type with `module Module exposing (UnionType(..), otherthing, etc)` and import it with `import Module exposing (UnionType(..))`.

**Comment by Gabgab2003** [comment]
> For anyone looking here. Export your union type with `module Module exposing (UnionType(..), otherthing, etc)` and import it with `import Module exposing (UnionType(..))`.

I somehow knew that importing it with (..) was possible, but I didnt think of exporting for some reason. ¯\_(ツ)_/¯

**Comment by tankorsmash** [comment]
This means that you can't access `Dog` and `Cat` without explicitly exposing them right? Just `import Submodule` isn't enough to access them it seems like.
